js/packages/mobile-wallet-adapter-protocol/src/createMobileWalletProxy.ts

```diff
@@ -1,175 +1,182 @@
-                }
-                case 'v1': {
-                    return {
-                        ...capabilities,
-                        supports_sign_and_send_transactions: true,
-                        supports_clone_authorization: capabilities.features.includes(SolanaCloneAuthorization)
-                    } as Awaited<ReturnType<MobileWallet[TMethodName]>>;
-                }
-            }
-        }
-    }
-    return response;
-};
-
-async function signInFallback(
-    signInPayload: SignInPayload,
-    authorizationResult: Awaited<ReturnType<MobileWallet['authorize']>>,
-    protocolRequestHandler: (method: string, params: Parameters<MobileWallet['signMessages']>[0]) => Promise<unknown>
-) {
-    const domain = signInPayload.domain ?? window.location.host;
-    const address = (authorizationResult as AuthorizationResult).accounts[0].address;
-    const siwsMessage = createSIWSMessageBase64({ ...signInPayload, domain, address })
-    const signMessageResult = await (protocolRequestHandler('sign_messages', 
-        { 
-            addresses: [ address ], 
-            payloads: [ siwsMessage ]
-        }
-    ) as ReturnType<MobileWallet['signMessages']>);
-    const signInResult: SignInResult = {
-        address: address,
-        signed_message: siwsMessage,
-        signature: signMessageResult.signed_payloads[0].slice(siwsMessage.length)
-    };
-    return signInResult;
+import { createSIWSMessageBase64 } from "./createSIWSMessage";
+import { 
+    AuthorizationResult, 
+    MobileWallet, 
+    ProtocolVersion, 
+    SignInPayload, 
+    SignInResult, 
+    SolanaCloneAuthorization, 
+    SolanaSignTransactions 
+} from "./types";
+import type { IdentifierArray } from "@wallet-standard/core";
+
+/**
+ * Creates a {@link MobileWallet} proxy that handles backwards compatibility and API to RPC conversion.
+ * 
+ * @param protocolVersion the protocol version in use for this session/request
+ * @param protocolRequestHandler callback function that handles sending the RPC request to the wallet endpoint.
+ * @returns a {@link MobileWallet} proxy
+ */
+export default function createMobileWalletProxy<
+    TMethodName extends keyof MobileWallet, 
+    TReturn extends Awaited<ReturnType<MobileWallet[TMethodName]>>
+>(
+    protocolVersion: ProtocolVersion,
+    protocolRequestHandler: (method: string, params: Parameters<MobileWallet[TMethodName]>[0]) => Promise<any>
+): MobileWallet {
+    return new Proxy<MobileWallet>({} as MobileWallet, {
+        get<TMethodName extends keyof MobileWallet>(target: MobileWallet, p: TMethodName) {
+            // Wrapping a Proxy in a promise results in the Proxy being asked for a 'then' property so must 
+            // return null if 'then' is called on this proxy to let the 'resolve()' call know this is not a promise.
+            // see: https://stackoverflow.com/a/53890904
+            //@ts-ignore
+            if (p === 'then') { 

```

Comment:
wtf git, weird diff. this if statement was the only thing added here. Thanks Mike for help tracking this issue down. the comment above explains why this is needed.